### ISSUE TITLE
Bug: "onboarding" screen black on initial deployment

### ISSUE THREAD

**Issue by voyager529** [TomBursch/kitchenowl#396]
### Is this urgent?

No

### What parts are affected

Frontend

### What is the server version

v0.4.21

### What is the client version

n/a - browser issue

### What platform are you using

Web

### What's the problem 🤔

I'll preface this by stating that this is almost definitely user error rather than a bug...but there was no category for that =).

Initial install of the Docker Compose was performed, per the documentation, changing only the exposed port, the volume, and the secret key.

When I go to the LAN IP:88, I get a "/onboarding" page, which seems to get the CSS code when I do a view source, but it's otherwise blank.

### Share your logs

```Shell
Nothing stood out to me in the logs (i.e. there were no errors shown in the logs of either container), but I'd be happy to share any logs on request.
```

### Share your configuration

```yml
version: "3"
services:
  front:
    image: tombursch/kitchenowl-web:latest
    restart: unless-stopped
    ports:
      - "88:80"
    depends_on:
      - back
  back:
    image: tombursch/kitchenowl:latest
    restart: always
    environment:
      - JWT_SECRET_KEY=REDACTED
    volumes:
      - nfs.kitchenowl:/data

volumes:
  nfs.kitchenowl:
```

**Comment by TomBursch** [comment]
Does this happen with the mobile apps too? What browser are you using, and are there any errors if you open the developer tools and look at the network requests on a fresh page load?

**Comment by voyager529** [comment]
Thanks so much for the ultra quick response, Tom! 

So, the answer actually was a relatively-obvious one, which was the 'try a different browser' idea. It wasn't working in Firefox, even in a private tab, but Opera worked just fine, and once I finished the onboarding, so did the app when on the LAN. So, there's some Firefox weirdness...I can definitely live with that (to answer your question about the network traffic, it _does_ pull some javascript files and fonts, so it's probably just an oddity with rendering for Gecko rather than Webkit).

If I might humbly request a bit more help, I can't seem to figure out my issue with my reverse proxy config. It's nginx based, and I thought I correctly copied your nginx config example and only changed the relevant parts, but I get  a 502 Bad Gateway when I attempt to get to it on the WAN. Here's my config:

```
server {
    server_name example.com;
    listen [::]:443 ssl ipv6only=on;
    listen 443 ssl;

    add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;

    location / {
        proxy_pass http://192.168.11.10:88/;
    }

    ssl_certificate /etc/nginx/ssl/example.com.crt;
    ssl_certificate_key /etc/nginx/ssl/example.com.key;
    ssl_dhparam /etc/nginx/dhparam.pem;
}
server {
    if ($host = example.com) {
        return 301 https://$host$request_uri;
    }

    server_name example.com;
    listen 80;
    listen [::]:80;
    return 404;
}
```

Thank you again for pointing me in the correct direction!

**Comment by voyager529** [comment]
Okay, I figured it out...sort of.

For whatever reason, my Docker implementation tends to be a bit 'special' when it comes to internal NATing...so I changed the network config to use my macvlan and give the 'front' and 'back' containers their own IPs, forwarding through port 80 and 5000 respectively. Upon doing this, I was able to get the reverse proxy to forward the traffic properly.

I'm sure there's a way more elegant solution than this...but I got it working and, once again, I'm super appreciative of the assistance.
